tests/listeners/test_empty_listener.py

```diff
@@ -0,0 +1,24 @@
+#
```

Comment:
Let's change the name of these files that don't contain tests to just `empty_listener` etc -- the `test_` prefix is confusing here as they don't contain any tests themselves, but are used by them.

Reply:
Done.